server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -294,6 +294,19 @@ public final class IndexSettings {
     }, new RefreshIntervalValidator(), Property.Dynamic, Property.IndexScope, Property.ServerlessPublic);
 
     static class RefreshIntervalValidator implements Setting.Validator<TimeValue> {
+
+        static final String STATELESS_ALLOW_INDEX_REFRESH_INTERVAL_OVERRIDE = "es.stateless.allow.index.refresh_interval.override";
+
+        private static class AllowOverrideLazyHolder {
+            private static final boolean ALLOW_OVERRIDE = Boolean.parseBoolean(
```

Comment:
nit Why the new class? You can put the static boolean directly inside the RefreshIntervalValidator class, right? Alternatively, you can use a LazyInitializable object instead.

Reply:
>You can put the static boolean directly inside the RefreshIntervalValidator class, right?

Yes and no
Yes from code simplicity point of view and No - from testability. The `allow_override` flag init happens before (at a time when jvm process class imports) a test manipulates with system properties and flag's init does not see updated sys props. So, in this case flag init should be either lazy or sys property should be read every time validation happened.

By using Inner static class jvm defer its initialization till the time it is actually used, in our use case to get wanted lazy behavior.  And big thanks for hint about `LazyInitializable ` I'll check it out

Reply:
I think this way of testing can introcuce flakiness depending on the order of test execution and which jvm it runs on. For example, the following fails for me locally.

```
./gradlew ':server:test' --tests "org.elasticsearch.index.IndexSettings*" -Dtests.jvms=1
```

The option `-Dtests.jvms=1` forces the two test classes, `IndexSettingsTests` and `IndexSettingsOverrideTests`, to run in the same test JVM. Definitely something possible on CI with enough runs. In this case, either `IndexSettingsTests#testStatelessMinRefreshInterval` or `IndexSettingsOverrideTests#testStatelessMinRefreshIntervalOverride` will fail depending on the order of execution.

I think we might need to exclude this new test class from the `test` class and [register a separate task](https://github.com/elastic/elasticsearch/blob/6f72d4ccaad1580bd8138948138b94bcb369dc95/modules/transport-netty4/build.gradle#L62) just for itself. We can then specify the system property in gradle build file which also helps replacing `LazyInitializable` with a simple static boolean field.

Reply:
Thanks @ywangd for this. I see your concern and raised [PR](https://github.com/elastic/elasticsearch/pull/110301) to improve things